### PR TITLE
LPS-195100 Add feedback when incorrect date has been changed

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
https://liferay.atlassian.net/browse/LPS-195100

When user enters incorrect year, the date is changed to the current date. This causes problems with accessibility as there is not sign that the date was changed.

I added feedback notification when the date is changed due to incorrect date. I also made sure the feedback is removed when the user changes the date correctly.

Let me know if there are any questions or comments about this.
Thank you!

Fix:
https://github.com/liferay-frontend/liferay-portal/assets/31672551/125a7522-694f-4396-aac2-284445c832fd

### CODE REVIEW

File: portal-web/docroot/html/taglib/ui/input_date/page.jsp [hunk 1]
```diff
@@ -121,7 +121,19 @@ else {
 					<aui:input autocomplete="<%= autoComplete %>" cssClass="<%= cssClass %>" disabled="<%= disabled %>" id="<%= HtmlUtil.getAUICompatibleId(name) %>" label="" name="<%= name %>" placeholder="<%= StringUtil.toLowerCase(placeholderValue) %>" required="<%= required %>" title="" type="text" useNamespace="<%= !StringPool.BLANK.equals(namespace) %>" value="<%= dateString %>" wrappedField="<%= true %>">
 						<aui:validator errorMessage="please-enter-a-valid-date" name="custom">
 							function(val) {
-								return AUI().use('aui-datatype-date-parse').Parsers.date('<%= mask %>', val);
+								var dateValidation = AUI().use('aui-datatype-date-parse').Parsers.date('<%= mask %>', val);
```
Comment (bryceosterhaus) [inline portal-web/docroot/html/taglib/ui/input_date/page.jsp:124]: You can use `const`

File: portal-web/docroot/html/taglib/ui/input_date/page.jsp [hunk 2]
```diff
@@ -121,7 +121,19 @@ else {
...
+
+								if (!dateValidation) {
+									var parentNode = A.one('#<%= nameId %>')._node.parentElement;
+
+									var feedbackElement = parentNode.querySelector('.form-feedback-item');
```
Comment (bryceosterhaus) [inline portal-web/docroot/html/taglib/ui/input_date/page.jsp:129]: `const` for both of these too

**Review by bryceosterhaus — COMMENTED** [review]
Looks good! Only think you'll want to change is using more modern js, such as const/let because those are supported in all modern browsers now

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, I pushed the changed. Let me know if there is anything else needed.
Thank you!

**Review by bryceosterhaus — APPROVED** [review]
LGTM!

**Comment by fortunatomaldonado** [comment]
ci:forward

**Comment by deborasoliveira** [comment]
I'm taking a look at the relevant results.

**Comment by deborasoliveira** [comment]
The failures are unrelated to the PR. 
- Some calendar tests failed but the failure started occurring before this PR, and their execution was successful when I ran them locally using this bundle.

**Comment by fortunatomaldonado** [comment]
ci:forward:force

**Comment by fortunatomaldonado** [comment]
ci:forward:force

**Comment by fortunatomaldonado** [comment]
FWD: https://github.com/brianchandotcom/liferay-portal/pull/140606

**Pull request closed without merging**